src/kv/test/kv_snapshot.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "kv/kv_serialiser.h"
+#include "kv/store.h"
+#include "kv/test/null_encryptor.h"
+#include "kv/tx.h"
+
+#include <doctest/doctest.h>
+
+struct MapTypes
+{
+  using StringString = kv::Map<std::string, std::string>;
+  using NumNum = kv::Map<size_t, size_t>;
+};
+
+TEST_CASE("Simple snapshot" * doctest::test_suite("snapshot"))
+{
+  kv::Store store;
+  auto& string_map = store.create<MapTypes::StringString>(
+    "string_map", kv::SecurityDomain::PUBLIC);
+  auto& num_map =
+    store.create<MapTypes::NumNum>("num_map", kv::SecurityDomain::PUBLIC);
+
+  kv::Version first_snapshot_version = kv::NoVersion;
+  kv::Version second_snapshot_version = kv::NoVersion;
+
+  INFO("Apply transactions to original store");
+  {
+    kv::Tx tx1;
+    auto view_1 = tx1.get_view(string_map);
+    view_1->put("foo", "bar");
+    REQUIRE(tx1.commit() == kv::CommitSuccess::OK);
+    first_snapshot_version = tx1.commit_version();
```

Comment:
This code is fine, but just adding a note because it confused me for a while: `Tx::commit_version()` is completely different from `Store::commit_version()`. The latter returns `compacted`, due to our usual confusion around local/global commit. We should do some kind of renaming, either make the latter `compact_version`, or change all of the 'local' `commit`s to `apply` or similar.

Reply:
Absolutely. I have raised https://github.com/microsoft/CCF/issues/1355 to do this work separately.